### PR TITLE
txn: move several functions to each command

### PR DESCRIPTION
**Pull request by longfangsong**
move several functions to each command

Signed-off-by: longfangsong <longfangsong@icloud.com>

### What problem does this PR solve?

Minor refactor in `src/txn/commands.rs`.

### What is changed and how it works?

Move `readonly` & `is_sys_cmd` & `requires_pessimistic_txn` & `can_pipelined` to `CommandExt` trait.

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: src/storage/txn/commands.rs [hunk 1]
```diff
@@ -278,23 +288,59 @@ macro_rules! command {
...
-            }
-            )?
+        fn incr_cmd_metric(&self) {
+            KV_COMMAND_COUNTER_VEC_STATIC.$tag.inc();
         }
-    }
+    };
+}
+
+macro_rules! readonly {
```
Comment (longfangsong) [inline src/storage/txn/commands.rs:314]: The following 4 macros seems so similar that I think we can abstract them into one.
But I cannot find a good name for this single macro (maybe `field!` ?).

**Comment by longfangsong** [comment]
@nrc PTAL.

File: src/storage/txn/commands.rs (hunk 1 above)
Comment (nrc) [reply src/storage/txn/commands.rs:314]: `command_method` or `cmd_method`?

File: src/storage/txn/commands.rs [hunk 2]
```diff
@@ -321,6 +365,11 @@ command! {
         }
 }
 
+impl CommandExt for Prewrite {
```
Comment (nrc) [inline src/storage/txn/commands.rs:368]: Just a syntactic nit, but do you think we could move these inside the `command` macro use?

File: src/storage/txn/commands.rs [hunk 3]
```diff
@@ -246,15 +246,25 @@ pub trait CommandExt {
     fn ts(&self) -> TimeStamp {
         TimeStamp::zero()
     }
+    fn readonly(&self) -> bool {
+        false
+    }
+    fn is_sys_cmd(&self) -> bool {
```
Comment (nrc) [inline src/storage/txn/commands.rs:252]: style: one line between each method please

File: src/storage/txn/commands.rs (hunk 2 above)
Comment (longfangsong) [reply src/storage/txn/commands.rs:368]: > Just a syntactic nit, but do you think we could move these inside the `command` macro use?

I’ve tried it, but things become tricky when handleing `self `.

**Comment by nrc** [comment]
@youjiali1995 PTAL

**Pull request merged**